### DISCUSSION TITLE
Forking all branches of repo for GitHub classroom

### DISCUSSION THREAD

**Discussion by naish-uwo** [community/community#101915] [GitHub Education]
### Hi everyone,

I have an assignment in which the students use multiple branches with different versions of the code. The template repository was set to be private and contained 6 branches. The forked student repos only get the main branch and are not able to see any of the others. I added myself as a student to confirm this behaviour. As a workaround, I've made the assignment repo public, so at least the students can clone it will all branches and copy over the code that they need.

Is there a setting that I don't have configured properly or does the new classroom not support forking of multiple branches? The old classroom model did work (but of course students couldn't pull any changes made to the template after they has cloned it). Any suggestions are appreciated.

Cheers,
Michael

Student repo (1 branch)
<img width="1550" alt="Screen Shot 2024-01-24 at 7 12 47 PM" src="https://github.com/community/community/assets/63020577/3d6eeadc-ac91-4a6f-aefb-7d98fff9ef3f">

Template repo (6 branches)
<img width="1555" alt="Screen Shot 2024-01-24 at 7 15 31 PM" src="https://github.com/community/community/assets/63020577/414d808b-ace5-4c6d-b626-6b295d9a8026">

**Comment by MaximeImbeau** [comment]
Same problem here!
